internal/keycloakb/onboardingmodule.go

```diff
@@ -102,3 +111,50 @@ func (om *onboardingModule) SendOnboardingEmail(ctx context.Context, accessToken
 
 	return nil
 }
+
+func (om *onboardingModule) generateUsername(chars []rune, length int) string {
+	var b strings.Builder
+
+	for j := 0; j < length; j++ {
+		nBig, _ := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		index := int(nBig.Int64())
+		b.WriteRune(chars[index])
+	}
+	return b.String()
+}
+
+func (om *onboardingModule) CreateUser(ctx context.Context, accessToken, realmName, targetRealmName string, kcUser *kc.UserRepresentation) (string, error) {
```

Comment:
I think it is error prone to have side effect on input parameters with methods returning something, especially in our code where we almost never do such thing. 
I would suggest to return (string, string, error) in this specific case. What do you think ?